site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js

```diff
@@ -220,27 +220,45 @@ exports.onCreatePage = ({ page, actions }, themeOptions) => {
 
     const [template, mdxFile] = page.component.split(`?__contentFilePath=`);
 
-    // if the mdxFile path possesses a language, let's strip the language to it
+    // If the mdxFile path has a language, let's strip the language from it.
     // ex: index.de.mdx ==> index.mdx
     if (mdxFile) {
-      //split the filename in three parts split by the dot.
-      let [thePath /* lang */, , ext] = mdxFile.split(`.`);
+      let thePath = path.dirname(mdxFile);
 
-      if (ext === `mdx`) {
-        //if there's data in the third part, just keep the first and last part, removing the language
-        theFilePath = `${thePath}.${ext}`;
-      } else {
-        //if there's no content in the third part, it means that there's no language part. No need to remove the language
+      let fileName = path.basename(mdxFile);
+
+      let ext = path.extname(mdxFile);
+
+      if (ext !== '.mdx') {
+        console.error(`Unexpected file format in mdx path parsing: ${mdxFile}`);
```

Comment:
Should we crash the build instead? Logs like this probably get ignored.

(or should be logged as warnings)

Reply:
Hard agree. What's the best way to crash here?

Reply:
Chose `throw` since there's not a specific gatsby hook (like I was expecting).